WordPress/Sniffs/WP/AlternativeFunctionsSniff.php

```diff
@@ -167,6 +166,44 @@ public function process_matched_token( $stackPtr, $group_name, $matched_content
 					return;
 				}
 
+				break;
+
+			case 'file_get_contents':
+				/*
+				 * Using `wp_remote_get()` will only work for remote URLs.
+				 * See if we can determine is this function call is for a local file and if so, bow out.
+				 */
+				$params = $this->get_function_call_parameters( $stackPtr );
+
+				if ( isset( $params[2] ) && 'true' === $params[2]['raw'] ) {
+					// Setting `$use_include_path` to `true` is only relevant for local files.
+					return;
+				}
+
+				if ( isset( $params[1] ) === false ) {
+					// If the file to get is not set, this is a non-issue anyway.
+					return;
+				}
+
+				if ( strpos( $params[1]['raw'], 'http:' ) !== false
+					|| strpos( $params[1]['raw'], 'https:' ) !== false
+				) {
+					// Definitely a URL, throw notice.
+					break;
+				}
+
+				if ( preg_match( '`\b(?:ABSPATH|WP_(?:CONTENT|PLUGIN)_DIR|WPMU_PLUGIN_DIR|TEMPLATEPATH|STYLESHEETPATH|(?:MU)?PLUGINDIR)\b`', $params[1]['raw'] ) === 1 ) {
```

Comment:
Feels like these regexes could be centralised for potential re-use, or the whole `preg_match()` could be inside a named local / parent sniff method to become a little more self-documenting.

Reply:
@GaryJones Good question.

IIRC, so far, we haven't needed anything like it elsewhere. Generally speaking, I'd move them to be more re-usable once a secondary use-case has been identified.

All the same, I'm not fussed about this, so what about adding them as class constants to the `Sniff` class and using the `x` modifier to spread them over several lines with documentation ?

Moving the whole function call to a boolean return function feels wrong as that would prevent sniffs which use the regexes from getting access to the `$matches` if so desired.

Reply:
Could the new method return the matches, and then the use case here cast to a bool if needed?

Reply:
> Could the new method return the matches, and then the use case here cast to a bool if needed?

The thing is `preg_match()` has two results:
* The function return value: `1` for a match, `0` for no match or `false` for an error/fault regex etc
* The `$matches` array which may or may not be filled with matches.

By returning just the `$matches`, we would loose the nuances the function return value provides.

I'm wondering if we're not trying to create a solution for a non-issue. Have you got a (future) sniff in mind which would need these regex(es) or an existing one which could benefit from using them ?

If not, let's leave the abstraction until the time when we do have a second use-case as that might also clarify what the best way to abstract this would be.